### PR TITLE
Make ForumMagnum more flexible so it's easier to fork new forums

### PR DESCRIPTION
**Pull request by michaelkeenan**
To make ForumMagnum more easily forkable for new forums, JP and I looked at every use of forumSetting, isEAForum, and isLW, and changed a lot of them to be more flexible. Some of them became database or instance settings, and others became an isFriendlyUI or isBookUI check, where the EA Forum (and the forked Waking Up Forum) are FriendlyUI forums, and LessWrong and the Alignment Forum are BookUI forums.

Database settings that will be needed before pushing this change:
hasLogo (default false)
	LW: false
	AF: false
	EA: true
	WU: true
communityNameSetting (default "Community")
	LW: "LessWrong Community"
	EA: "EA Community"
showCurated (default false)
	LW: true
	AF: false
	EA: true
showCommunityMapSetting (default false)
	EA: true
showThemePickerSetting (default false)
	LW: true
	AF: true
	EA: false
hasShareButtonSetting (default true)
	LW: true
	AF: true
	EA: true
verifyEmailsSetting (default true)
	EA: false
	LW: true
	AF: true
	WU: false
adminTeamEmailUsernameSetting (default "ERROR: NO ADMIN TEAM EMAIL USERNAME SET")
	EA: "AdminTeam"
	LW: "LessWrong"
adminTeamEmailUsernameSetting (default "ERROR: NO ADMIN TEAM EMAIL ADDRESS SET")
	EA: "forum@effectivealtruism.org"
	LW: "team@lesswrong.com"
mainThemeColorSetting (default "#5f9b65")
	EA: "#0C869B"
	LW: "#5f9b65"
	AF: "#5f9b65"

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205586840652993) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/events/modules/EventCards.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/events/modules/EventCards.tsx:10]: `isLW` is unused here

**Review by Will-Howard — COMMENTED** [review]
I've reviewed about half of this, posting the comments no so you can get started on addressing them. I'll try to get to the other half either later today or tomorrow.

None of the comments are serious, the overall approach seems good 👍

File: packages/lesswrong/components/recommendations/ConfigurableRecommendationsList.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/recommendations/ConfigurableRecommendationsList.tsx:37]: This `zzz` is accidental presumably?

File: packages/lesswrong/components/shortform/NewShortformDialog.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/shortform/NewShortformDialog.tsx:34]: What do these do?

File: packages/lesswrong/server/forumSpecificMiddleware/eaMiddleware.ts
Comment (Will-Howard) [inline packages/lesswrong/server/forumSpecificMiddleware/eaMiddleware.ts:6]: I assume this is needed for WU, but not for any of the existing sites? I think I would rather remove this functionality in that case, and just have it in the WU fork

File: packages/lesswrong/lib/publicSettings.ts
Comment (Will-Howard) [inline packages/lesswrong/lib/publicSettings.ts:142]: Could you do all these before merging?

File: packages/lesswrong/lib/publicSettings.ts
Comment (Will-Howard) [inline packages/lesswrong/lib/publicSettings.ts:150]: Do you mean, it would be better as an instance setting than a db setting?

File: packages/lesswrong/lib/cookies/utils.ts
Comment (Will-Howard) [inline packages/lesswrong/lib/cookies/utils.ts:4]: unused

**Review by Will-Howard — APPROVED** [review]
Approved subject to comments 👍. Although could you please hold off on merging until we've all set up the new settings

**Comment by Will-Howard** [comment]
This is my estimate of what we need to add to our database settings ( @michaelkeenan does this look right? Some of the settings were different to your comment):
EA:
```
module.exports.databasePublicSettings = {
  communityName: "EA Community",
  hasLogo: true,
  showCurated: true,
  showCommunityMap: true,
  hasShareButton: true,
  verifyEmails: false,
  adminAccount: { username: "AdminTeam", email: "forum@effectivealtruism.org" },
  mainThemeColor: "#0C869B"
}
```
LW:
```
module.exports.databasePublicSettings = {
  communityName: "LessWrong Community",
  hasLogo: false,
  showCurated: true,
  showCommunityMap: false,
  hasShareButton: true,
  verifyEmails: true,
  adminAccount: { username: "LessWrong", email: "team@lesswrong.com" },
  mainThemeColor: "#5f9b65"
}
```
AF:
```
module.exports.databasePublicSettings = {
  hasLogo: false,
  showCurated: false,
  showCommunityMap: false,
  hasShareButton: true,
  verifyEmails: true,
  mainThemeColor: "#5f9b65"
}
```

**Comment by b0b3rt** [comment]
@michaelkeenan overall looks good to me, though it does look like there are a few TODOs left to finish up (noted by @Will-Howard).  We do have the slight problem that we can't have different database settings between LW and AF, since they point to the same database instance - in cases where those values are different, we probably need those to be instance settings instead.  I think that's only `showCurated`.  `adminAccount` probably needs to exist for both LW and AF, though I'm not sure if it ever gets used in an AF context in practice.  Maybe for comment deletion?

Please give me a heads-up when this is ready to go so I can set the finalized database & instance settings before it's merged.

**Comment by michaelkeenan** [comment]
Thanks for looking at this! I've had various things come up, but expect to get back to this on Wednesday.

File: .gitignore
Comment (jpaddison3) [reply .gitignore:26]: This is because you might want to have `local-db-settings.js` in your project directory and not commit it. Broadly, although we have `../ForumCredentials`, the intended way to start developing on this repo is to run with your settings in the project directory.

File: packages/lesswrong/components/comments/CommentPermalink.tsx
Comment (jpaddison3) [reply packages/lesswrong/components/comments/CommentPermalink.tsx:119]: I believe the divider is the LW compass rose.

File: packages/lesswrong/components/shortform/NewShortformDialog.tsx
Comment (jpaddison3) [reply packages/lesswrong/components/shortform/NewShortformDialog.tsx:34]: I believe the answer is to prevent someone from losing a bunch of their writing because they accidentally clicked outside of the modal.

File: packages/lesswrong/server/forumSpecificMiddleware/eaMiddleware.ts
Comment (jpaddison3) [reply packages/lesswrong/server/forumSpecificMiddleware/eaMiddleware.ts:6]: I like this pattern. I would like to see us move in that direction for code quality and ease-of-stamping reasons.

**Comment by jpaddison3** [comment]
JP's new list for settings that need to be updated:

```
EA:
instanceSettings:
verifyEmails: false
hasCuratedPosts: true

databasePublicSettings:
defaultSequenceBannerId: Banner/yeldubyolqpl3vqqy0m6.jpg
hasProminentLogo: true
adminAccount: { username: "AdminTeam", email: "forum@effectivealtruism.org" },

LW:
instanceSettings:
hasCuratedPosts: true

databasePublicSettings:
defaultSequenceBannerId: sequences/vnyzzznenju0hzdv6pqb.jpg
adminAccount: { username: "LessWrong", email: "team@lesswrong.com" },

AF:
N/A

WU:
hasProminentLogo: true
hasShareButtons: false
```

**Comment by jpaddison3** [comment]
@Will-Howard could you take another look please?

File: packages/lesswrong/components/common/Header.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/common/Header.tsx:17]: TODO for EAF and LW: change the name of this setting

File: packages/lesswrong/components/questions/AnswersSorting.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/questions/AnswersSorting.tsx:9]: Is this something you want to do before merging?

File: packages/lesswrong/server/emails/renderEmail.tsx
Comment (Will-Howard) [inline packages/lesswrong/server/emails/renderEmail.tsx:75]: Have you tested this by sending an email?

File: packages/lesswrong/components/sequences/SequencesPage.tsx
Comment (Will-Howard) [inline packages/lesswrong/components/sequences/SequencesPage.tsx:143]: Nice

**Review by Will-Howard — COMMENTED** [review]
The updated version looks good! I'm happy for us to merge this once the latest conflicts have been resolved and we and LW have added the appropriate settings

In order to make sure the settings don't get messed up in the switchover I'm planning to write a script to do a diff of the settings we have vs all the ones that are allowed. I'll do this after the conflicts have been fixed

(cc @b0b3rt for LW)

File: packages/lesswrong/components/questions/AnswersSorting.tsx
Comment (michaelkeenan) [reply packages/lesswrong/components/questions/AnswersSorting.tsx]: Upon looking at this, commentViewOptions seemed the right one to use, with answers being more like comments than they are like posts. I've made the change, which just required a re-ordering of the default comment options. I hope what I did makes sense.

File: packages/lesswrong/server/emails/renderEmail.tsx
Comment (michaelkeenan) [reply packages/lesswrong/server/emails/renderEmail.tsx:75]: I'm having trouble persuading my dev setup to send emails— @jpaddison3, might it be easy for you to check this? If not, I can probably get to this tomorrow.

Comment (jpaddison3) [reply packages/lesswrong/server/emails/renderEmail.tsx:75]: I tested it in the email previewer, IIRC. I'll test it now.

... and it's good.

**Comment by jpaddison3** [comment]
I think this is now a deployment question. @b0b3rt ready?

**Review by b0b3rt — APPROVED** [review]

**Comment by jimrandomh** [comment]
This PR broke the ability to request confirmation email resending on LW. @michaelkeenan

**Comment by michaelkeenan** [comment]
Ah, good to know; sorry about that

**Pull request merged**
